### ISSUE TITLE
Pluggy IndexError when running local development server

### ISSUE THREAD

**Issue by anlutro** [devpi/devpi#458]
Following the instructions here: http://doc.devpi.net/latest/quickstart-pypimirror.html

The only difference is that I've `git clone`d the repository and done `pip install -e server` inside a virtualenv to install dependencies. Functionally this should be the same as `pip install devpi-server`.

However, when running the command `devpi-server`, I get this error:

```
2017-08-29 11:46:35,188 INFO  NOCTX Loading node info from /home/andreas/.devpi/server/.nodeinfo
2017-08-29 11:46:35,189 INFO  NOCTX wrote nodeinfo to: /home/andreas/.devpi/server/.nodeinfo
Traceback (most recent call last):
  File "/home/andreas/code/devpi/server/.venv/bin/devpi-server", line 11, in <module>
    load_entry_point('devpi-server', 'console_scripts', 'devpi-server')()
  File "/home/andreas/code/devpi/server/devpi_server/main.py", line 56, in main
    return _main(pluginmanager, argv=argv)
  File "/home/andreas/code/devpi/server/devpi_server/main.py", line 126, in _main
    return xom.main()
  File "/home/andreas/code/devpi/server/devpi_server/main.py", line 230, in main
    res = xom.config.hook.devpiserver_cmdline_run(xom=xom)
  File "/home/andreas/code/devpi/server/.venv/lib/python3.5/site-packages/pluggy/__init__.py", line 680, in __call__
    return self._hookexec(self, self._nonwrappers + self._wrappers, kwargs)
  File "/home/andreas/code/devpi/server/.venv/lib/python3.5/site-packages/pluggy/__init__.py", line 240, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "/home/andreas/code/devpi/server/.venv/lib/python3.5/site-packages/pluggy/__init__.py", line 234, in <lambda>
    methods, kwargs, specopts=hook.spec_opts, hook=hook
  File "/home/andreas/code/devpi/server/.venv/lib/python3.5/site-packages/pluggy/callers.py", line 110, in execute
    return outcome.get_result()[0]
IndexError: list index out of range
```

Version info:

```
$ python --version
Python 3.5.3
$ devpi-server --version
4.3.1.dev0
```

**Comment by RonnyPfannschmidt** [comment]
what version of pluggy?

**Comment by anlutro** [comment]
`0.5.0`. Here's my full list of pip dependencies in case others are relevant as well:

```
$ pip freeze | sort
apipkg==1.4
argon2-cffi==16.3.0
certifi==2017.7.27.1
cffi==1.10.0
chardet==3.0.4
devpi-common==3.1.0
-e git+https://github.com/devpi/devpi@9362335198c408fa14a32cd97dacdd99d9a94b64#egg=devpi_server&subdirectory=server
execnet==1.4.1
hupper==1.0
idna==2.6
itsdangerous==0.24
passlib==1.7.1
PasteDeploy==1.5.2
pkg-resources==0.0.0
plaster==0.5
plaster-pastedeploy==0.4.1
pluggy==0.5.0
py==1.4.34
pycparser==2.18
pyramid==1.9.1
repoze.lru==0.6
requests==2.18.4
six==1.10.0
translationstring==1.3
urllib3==1.22
venusian==1.1.0
waitress==1.0.2
WebOb==1.7.3
zope.deprecation==4.3.0
zope.interface==4.4.2
```

**Comment by RonnyPfannschmidt** [comment]
this regression was introduced in pluggy 0.5 - please install `pluggy==0.4.0` we will work on a fix

**Comment by tgoodlet** [comment]
@RonnyPfannschmidt @anlutro fix is in pytest-dev/pluggy#69

There's a new test that should replicate this but it would be handy if someone could test that branch just to be double sure 😃

**Comment by fschulze** [comment]
The pluggy 0.5.1 release is out and fixes this issue.
